src/aliceVision/keyframe/KeyframeSelector.cpp

```diff
@@ -30,448 +33,383 @@ namespace keyframe {
+            bl = sumsq.at<double>(i + window_size, j);
+            br = sumsq.at<double>(i + window_size, j + window_size);
+            double s2 = br + tl - tr - bl;
 
-    const auto& feed = *_feeds.back();
+            double std_2 = sqrt((s2 - (s1 * s1) / n) / n);
 
-    // check if feed is initialized
-    if(!feed.isInit())
-    {
-      ALICEVISION_LOG_ERROR("Cannot initialize the FeedProvider with " << path);
-      throw std::invalid_argument("Cannot while initialize the FeedProvider with " + path);
+            maxstd = std::max(maxstd, std_2);
+        }
     }
 
-    // update minimum number of frames
-    nbFrames = std::min(nbFrames, feed.nbFrames() - static_cast<std::size_t>( _cameraInfos.at(mediaIndex).frameOffset));
-  }
-
-  // check if minimum number of frame is zero
-  if(nbFrames == 0)
-  {
-    ALICEVISION_LOG_ERROR("One or multiple medias can't be found or empty !");
-    throw std::invalid_argument("One or multiple medias can't be found or empty !");
-  }
-
-  // resize selection data vector
-  _framesData.resize(nbFrames);
-
-  // feed provider variables
-  image::Image< image::RGBColor> image;    // original image
-  camera::PinholeRadialK3 queryIntrinsics; // image associated camera intrinsics
-  bool hasIntrinsics = false;              // true if queryIntrinsics is valid
-  std::string currentImgName;              // current image name
-
-  // process variables
-  const unsigned int frameStep = _maxFrameStep - _minFrameStep;
-  const unsigned int tileSharpSubset = (_nbTileSide * _nbTileSide) / _sharpSubset;
-
-  // create output folders
-  if(_feeds.size() > 1)
-  {
-    const std::string rigFolder = _outputFolder + "/rig/";
-    if(!fs::exists(rigFolder))
-      fs::create_directory(rigFolder);
-
-    for(std::size_t mediaIndex = 0 ; mediaIndex < _feeds.size(); ++mediaIndex)
+    return maxstd;
+}
+
+cv::Mat readImage(dataio::FeedProvider & feed, size_t max_width)
+{
+    image::Image<image::RGBColor> image;
+    camera::PinholeRadialK3 queryIntrinsics;
+    bool hasIntrinsics = false;
+    std::string currentImgName;
+
+    if (!feed.readImage(image, queryIntrinsics, currentImgName, hasIntrinsics))
     {
-      const std::string subPoseFolder = rigFolder + std::to_string(mediaIndex);
-      if(!fs::exists(subPoseFolder))
-        fs::create_directory(subPoseFolder);
+        ALICEVISION_LOG_ERROR("Cannot read frame '" << currentImgName << "' !");
+        throw std::invalid_argument("Cannot read frame '" + currentImgName + "' !");
     }
-  }
-  
-  // feed and metadata initialization
-  for(std::size_t mediaIndex = 0 ; mediaIndex < _feeds.size(); ++mediaIndex)
-  {
-    // first frame with offset
-    _feeds.at(mediaIndex)->goToFrame(_cameraInfos.at(mediaIndex).frameOffset);
-
-    if(!_feeds.at(mediaIndex)->readImage(image, queryIntrinsics, currentImgName, hasIntrinsics))
+
+    //Convert content to opencv
+    cv::Mat cvFrame(cv::Size(image.cols(), image.rows()), CV_8UC3, image.data(), image.cols() * 3);
```

Comment:
imageRGBAToCvMatBGR

Reply:
The feed's output (image::RGBColor) doesn't have the right format for imageRGBAToCvMatBGR (which expects image::RGBAfColor).